files/image_config/kubernetes/kubelet/10-kubeadm.conf

```diff
@@ -0,0 +1,11 @@
+# Note: This dropin only works with kubeadm and kubelet v1.11+
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/default/kubelet
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_CONFIG_ARGS $KUBELET_KUBEADM_ARGS $KUBELET_EXTRA_ARGS
```

Comment:
This file is crowded and difficult to read.
Could you explain what is the minimal required here and why?

Reply:
This is as instructed by Kubernetes folks.  I just made it less crowded.

Reply:
Now readable. Do you have a reference link to explain why they are required? I guess we could remove all of them.

---
In reply to: [404429220](https://github.com/Azure/sonic-buildimage/pull/4374#discussion_r404429220) [](ancestors = 404429220)

Reply:
File removed, as it gets auto installed as part of kubeadm package install. Tested to work.